pkg/pdatagen/pmetricsgen/data.go

```diff
@@ -24,105 +50,92 @@ func Generate(opts ...GenerationOption) pmetric.Metrics {
 	}
 
 	metrics := pmetric.NewMetrics()
-	resourceMetrics := metrics.ResourceMetrics().AppendEmpty()
-	for i := 0; i < generationOpts.resourceAttributeCount; i++ {
-		suffix := strconv.Itoa(i)
-		// Do not change the key name format in resource attributes below.
-		resourceMetrics.Resource().Attributes().PutStr("resource."+suffix, generationOpts.resourceAttributeStringValue)
-	}
+	_, sm := setupResourceAndScope(&metrics,
+		generationOpts.resourceAttributeCount,
+		generationOpts.scopeAttributeCount,
+		generationOpts.resourceAttributeStringValue,
+		generationOpts.scopeAttributeStringValue)
 
-	scopeMetrics := resourceMetrics.ScopeMetrics().AppendEmpty()
-	for i := 0; i < generationOpts.scopeAttributeCount; i++ {
-		suffix := strconv.Itoa(i)
-		// Do not change the key name format in resource attributes below.
-		scopeMetrics.Scope().Attributes().PutStr("scope."+suffix, generationOpts.scopeAttributeStringValue)
-	}
+	// Pre-allocate capacity for all metrics
+	totalMetrics := generationOpts.count.GaugeMetricsCount +
+		generationOpts.count.SumMetricsCount +
+		generationOpts.count.HistogramMetricsCount +
+		generationOpts.count.ExponentialHistogramMetricsCount +
+		generationOpts.count.SummaryMetricsCount
+	sm.Metrics().EnsureCapacity(totalMetrics)
 
-	scopeMetrics.Metrics().EnsureCapacity(
-		generationOpts.count.GaugeMetricsCount +
-			generationOpts.count.SumMetricsCount +
-			generationOpts.count.HistogramMetricsCount +
-			generationOpts.count.ExponentialHistogramMetricsCount,
-	)
-
-	for i := 0; i < generationOpts.count.GaugeMetricsCount; i++ {
-		gaugeMetric := scopeMetrics.Metrics().AppendEmpty().SetEmptyGauge()
-		gaugeMetric.DataPoints().EnsureCapacity(generationOpts.count.GaugeDataPointCount)
-		for j := 0; j < generationOpts.count.GaugeDataPointCount; j++ {
-			dp := gaugeMetric.DataPoints().AppendEmpty()
-			dp.SetTimestamp(pcommon.NewTimestampFromTime(time.Now()))
-			dp.SetDoubleValue(rand.Float64())
-		}
-	}
+	// Generate and copy metrics
+	copyMetricsToScope(GenerateGaugeMetrics(
+		generationOpts.count.GaugeMetricsCount,
+		generationOpts.count.GaugeDataPointCount,
+		0,
```

Comment:
Add `generationOptions.pointAttributeCount` to generationOptions and use it everywhere instead random zero, which is more readable and configurable. This random zero breaks code reading flow and one has to read GenerateGaugeMetrics signature to unerstand what the zero maps to.